skimage/_shared/utils.py

```diff
@@ -107,6 +145,190 @@ def fixed_func(*args, **kwargs):
+        stop_version,
+        template=None,
+        new_name=None,
+        modify_docstring=True,
+        stacklevel=None,
+    ):
+        self.deprecated_name = deprecated_name
+        self.new_name = new_name
+        self.template = template
+        self.start_version = start_version
+        self.stop_version = stop_version
+        self.modify_docstring = modify_docstring
+        self.stacklevel = stacklevel
+
+    def __call__(self, func):
+        parameters = inspect.signature(func).parameters
+        deprecated_idx = list(parameters.keys()).index(self.deprecated_name)
+        if self.new_name:
+            new_idx = list(parameters.keys()).index(self.new_name)
+        else:
+            new_idx = False
+
+        if parameters[self.deprecated_name].default is not DEPRECATED:
+            raise RuntimeError(
+                f"Expected `{self.deprecated_name}` to have the value {DEPRECATED!r} "
+                f"to indicate its status in the rendered signature."
+            )
+
+        if self.template is not None:
+            template = self.template
+        elif self.new_name is not None:
+            template = self.replace_parameter_template
+        else:
+            template = self.remove_parameter_template
+        warning_message = template.format(
+            deprecated_name=self.deprecated_name,
+            deprecated_version=self.start_version,
+            changed_version=self.stop_version,
+            func_name=func.__qualname__,
+            new_name=self.new_name,
+        )
+
+        @functools.wraps(func)
+        def fixed_func(*args, **kwargs):
+            deprecated_value = DEPRECATED
+            new_value = DEPRECATED
+
+            # Extract value of deprecated parameter
+            if len(args) > deprecated_idx:
+                deprecated_value = args[deprecated_idx]
+                args = (
+                    args[:deprecated_idx] + (DEPRECATED,) + args[deprecated_idx + 1 :]
+                )
+            if self.deprecated_name in kwargs.keys():
+                deprecated_value = kwargs[self.deprecated_name]
+                kwargs[self.deprecated_name] = DEPRECATED
+            # Extract value of new parameter (if present)
+            if new_idx is not False and len(args) > new_idx:
+                new_value = args[new_idx]
+            if self.new_name and self.new_name in kwargs.keys():
+                new_value = kwargs[self.new_name]
+
+            if deprecated_value is not DEPRECATED:
+                stacklevel = (
+                    self.stacklevel
+                    if self.stacklevel is not None
+                    else _warning_stacklevel(func)
+                )
+                warnings.warn(
+                    warning_message, category=FutureWarning, stacklevel=stacklevel
+                )
+
+                if new_value is not DEPRECATED:
+                    raise ValueError(
+                        f"both deprecated parameter `{self.deprecated_name}` "
```

Comment:
```suggestion
                        f"Both deprecated parameter `{self.deprecated_name}` "
```

Reply:
As this is not a multi-sentence message I would default to the practice that Python itself uses: start error messages with lower case.

Reply:
Oh, good to know! I wasn't aware of this rule. Well, extending the spirit of my ':' suggestion, I would prefer this message in a two-sentence version, where "Please use only the latter ..." starts a second sentence.

Reply:
Done. :)